src/test/java/roomescape/reservation/repository/ReservationRepositoryTest.java

```diff
@@ -0,0 +1,133 @@
+package roomescape.reservation.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import roomescape.reservation.entity.Reservation;
+
+@SpringBootTest
+class ReservationRepositoryTest {
+
+    @Autowired
+    private ReservationRepository reservationRepository;
+
+    @DisplayName("초기 ID의 값은 1이다.")
+    @Test
+    void test1() {
+        long result = reservationRepository.generateId();
+
+        assertThat(result).isEqualTo(1);
+    }
+
+    @DisplayName("ID의 값은 순차적으로 올라간다.")
+    @Test
+    void test2() {
+        for (int i = 1; i <= 5000; i++) {
+            long result = reservationRepository.generateId();
+
+            assertThat(result).isEqualTo(i);
+        }
+    }
+
+    @DisplayName("예약 정보를 저장한다.")
+    @Test
+    void test3() {
+        // given
+        LocalDateTime now = LocalDateTime.now();
+        Reservation reservation = new Reservation(1, "꾹", now);
+
+        // when
+        Reservation result = reservationRepository.save(reservation);
+
+        // then
+        assertThat(result).isEqualTo(reservation);
+    }
+
+    @DisplayName("id가 같다면 해당 예약 정보로 변경한다.")
+    @Test
+    void test4() {
+        // given
+        LocalDateTime now = LocalDateTime.now();
+
+        String originalName = "꾹";
+        String changedName = "드라고";
+        Reservation reservation = new Reservation(1, originalName, now);
+        reservationRepository.save(reservation);
+
+        Reservation updateReservation = new Reservation(1, changedName, now);
+
+        // when
+        Reservation result = reservationRepository.save(updateReservation);
+
+        // then
+        assertThat(result).isEqualTo(updateReservation);
+    }
+
+    @DisplayName("id로 예약 정보를 가져온다")
+    @Test
+    void test5() {
+        // given
+        long id = 2;
+        LocalDateTime now = LocalDateTime.now();
+        Reservation reservation = new Reservation(id, "꾹", now);
+        reservationRepository.save(reservation);
+
+        // when
+        Optional<Reservation> result = reservationRepository.findById(id);
+
+        // then
+        assertThat(result).contains(reservation);
+    }
+
+    @DisplayName("모든 예약 정보를 가져온다.")
+    @Test
+    void test6() {
+        // given
+        LocalDateTime now = LocalDateTime.now();
+
+        Reservation reservation1 = new Reservation(1, "꾹", now);
+        Reservation reservation2 = new Reservation(2, "꾹", now);
+        Reservation reservation3 = new Reservation(3, "꾹", now);
+
+        List<Reservation> reservations = List.of(reservation1, reservation2, reservation3);
```

Comment:
각 요소를 개별로 사용하지 않으니, 변수를 분리해두지 않아도 괜찮아 보여요~

```suggestion
        List<Reservation> reservations = List.of(
                new Reservation(1, "꾹", now),
                new Reservation(2, "꾹", now),
                new Reservation(3, "꾹", now)
        );
```